internal/oci/uvm.go

```diff
@@ -336,6 +357,10 @@ func SpecToUVMCreateOpts(ctx context.Context, s *specs.Spec, id, owner string) (
 		lopts.StorageQoSBandwidthMaximum = ParseAnnotationsStorageBps(ctx, s, annotationStorageQoSBandwidthMaximum, lopts.StorageQoSBandwidthMaximum)
 		lopts.StorageQoSIopsMaximum = ParseAnnotationsStorageIops(ctx, s, annotationStorageQoSIopsMaximum, lopts.StorageQoSIopsMaximum)
 		lopts.PreferredRootFSType = parseAnnotationsPreferredRootFSType(ctx, s.Annotations, annotationPreferredRootFSType, lopts.PreferredRootFSType)
+		lopts.DefaultNUMA = parseAnnotationsBool(ctx, s.Annotations, annotationDefaultNUMA, lopts.DefaultNUMA)
+		if v, ok := s.Annotations[annotationVirtualNodeSettings]; ok {
+			lopts.NUMATopologyJSON = v
```

Comment:
I'm not a big fan of dumping a JSON blob into an annotation. What about using multiple annotations, where the name contains the index of the virtual NUMA node?

e.g.:
```
io.microsoft.virtualmachine.computetopology.numa.virtualnodes.0.physicalnode = 1
io.microsoft.virtualmachine.computetopology.numa.virtualnodes.0.processorcount = 32
```

Reply:
I thought we had discussed this internally and decided to go with JSON. But then, I thought the suggestion was an annotation specifying a separate file, which reportedly is how Kata handles this kind of thing.

Reply:
@jstarks correct this is what we'd like to move to but there wasn't a plan to do that in this change specifically. We all need to get together and come up with a schema we'd like to expose for the one JSON file.

Reply:
Yes, we want to move to a JSON file pointed to by an annotation. I just didn't want to block this PR on determining a schema for that file.

Reply:
and @kevpar I can give the suggestion a go and see how that turns out